File: 1. Practica/1. Introduccion-POO/ejercicio-3/Producto.py
class Producto:
    def __init__(self, nombre, precio, stock):
        self.nombre = nombre
        self.precio = precio
        self.stock = stock

    def vender(self, cantidad):
        if cantidad <= self.stock:
            self.stock -= cantidad
            print(f"Venta realizada: {cantidad} u")
        else:
            print(f"No hay suficiente stock para vender {cantidad} u")

    def reabastecer(self, cantidad):
        self.stock += cantidad
        print(f"Restablecer: {cantidad} u")


c = Producto("CARNE", 50.5, 100)
print("Producto:", c.nombre)
print("Precio:", c.precio, "Bs.")
print("Stock:", c.stock)

c.vender(50)
c.reabastecer(7)

print("Producto:", c.nombre)
print("Precio:", c.precio, "Bs.")
print("Stock:", c.stock)

File: 1. Practica/1. Introduccion-POO/ejercicio-3/Producto.java
package Introduccion_POO;
public class Producto {
    public String nombre;
    public double precio;
    public int stock;
    public Producto(String nombre, double precio, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
    }
    public void vender(int cantidad) {
        if (cantidad <= stock) {
            stock -= cantidad;
            System.out.println("Venta realizada: " + cantidad + " u");
        } else {
            System.out.println("No hay suficiente stock para vender " + cantidad + " u");
        }
    }
    public void reabastecer(int cantidad) {
        stock += cantidad;
        System.out.println("restablecer: " + cantidad + " u");
    }
}


